CMakeLists.txt

```diff
@@ -314,6 +314,7 @@ if(AVIF_ENABLE_WERROR)
 endif()
 
 if(AVIF_ENABLE_NODISCARD)
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
This is too early, before the following code at line 642:

```
if(AVIF_USE_CXX)
    enable_language(CXX)
    set(CMAKE_CXX_STANDARD 14)
endif()
```

It seems that we should change that code to:

```
if(AVIF_USE_CXX)
    enable_language(CXX)
    if(AVIF_ENABLE_NODISCARD)
        set(CMAKE_CXX_STANDARD 17)
    else()
        set(CMAKE_CXX_STANDARD 14)
    endif()
endif()
```